packages/shared/client/ReplayClient.ts

```diff
@@ -928,6 +776,32 @@ export class ReplayClient implements ReplayClientInterface {
     });
   }
 
+  streamAnalysis(
+    params: AnalysisParams,
+    handlers: {
+      onPoints?: (points: PointDescription[]) => void;
+      onResults?: (results: AnalysisEntry[]) => void;
+      onError?: (error: any) => void;
+    }
+  ): { pointsReceived: Promise<void>; resultsReceived: Promise<void> } {
+    let resolvePointsReceived!: () => void;
+    const pointsReceived = new Promise<void>(resolve => (resolvePointsReceived = resolve));
+    let resolveResultsReceived!: () => void;
+    const resultsReceived = new Promise<void>(resolve => (resolveResultsReceived = resolve));
+
+    this._waitForRangeToBeLoaded(params.range || null).then(() =>
+      analysisManager.runAnalysis(params, {
+        onAnalysisPoints: handlers.onPoints,
+        onAnalysisResult: handlers.onResults,
```

Comment:
Naming nit: Why is a param called "on results" (plural) re-mapped to something called "on result" (singular)?